src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -84,6 +69,29 @@ public void Move(int oldIndex, int newIndex)
         {
             MoveItem(oldIndex, newIndex);
         }
+        
+        /// <summary>
+        /// Adds a range of objects but does not raise the OnCollectionChanged event until all objects are added.
+        /// </summary>
+        /// <param name="collection">The set of objects to be added.</param>
+        /// <param name="clearFirst">Determines whether or not to remove any existing items in the collection first. Defaults to False.</param>
+        public void AddRange(IEnumerable<T> collection, bool clearFirst = false)
+        {
+            CheckReentrancy();
+
+            if (clearFirst) Items.Clear();
+
+            var newIndex = Items.Count;
+            foreach (var item in collection)
+            {
+                base.InsertItem(newIndex, item);
+                newIndex++;
+            }
```

Comment:
> Regarding the typecheck, the internal Items IList for Collection.cs is a concrete List<T>, the new instance of which is created on Line 25. So I would think your code optimization would still apply.

That's true for the default .ctor, but there's another .ctor ([line 29](https://github.com/dotnet/coreclr/blob/a0a055ba3cf8265055a37a14f206e9e30836bc18/src/mscorlib/src/System/Collections/ObjectModel/Collection.cs#L29-L34)) that takes any `IList<T>`.